libs/execution_context/ebpf_program.c

```diff
@@ -550,3 +573,29 @@ ebpf_program_free_program_info(_In_opt_ _Post_invalid_ ebpf_program_info_t* prog
         ebpf_free(program_info);
     }
 }
+
+void
+ebpf_program_attach_link(_Inout_ ebpf_program_t* program, _Inout_ ebpf_link_t* link)
+{
+    // Acquire "attach" reference on the link object.
+    ebpf_object_acquire_reference((ebpf_object_t*)link);
+
+    // Insert the link in the attach list.
+    ebpf_lock_state_t state;
+    state = ebpf_lock_lock(&program->links_lock);
+    ebpf_list_insert_tail(&program->links, &((ebpf_object_t*)link)->list_entry);
```

Comment:
should we test that list_entry is empty before inserting? Fail the call otherwise. In that case attach_link should return an error. Or at least assert.

Reply:
I think we dont need to check (and return error) in this case. This is an internal API, so caller can be trusted. There is also a check in `ebpf_link_attach_program()` that link->program should be NULL before calling ebpf_program_attach_link().

We can add an assert though.